compiler/rustc_hir_typeck/messages.ftl

```diff
@@ -45,6 +45,7 @@ hir_typeck_convert_using_method = try using `{$sugg}` to convert `{$found}` to `
 hir_typeck_ctor_is_private = tuple struct constructor `{$def}` is private
 
 hir_typeck_dependency_on_unit_never_type_fallback = this function depends on never type fallback being `()`
+    .note = in edition 2024, the requirement `{$obligation}` will fail
```

Comment:
maybe we can mention somehow that we plan to break this on all editions eventually?

Reply:
do you have an opinion on the wording?

```suggestion
    .note = in edition 2024, the requirement `{$obligation}` will fail, and eventually this will fail everywhere
```

seems awkward, and I have no idea what a better wording could be 🤔